### PR TITLE
GPU and TPU usage guide

### LINKED ISSUE #808: [BUG] TypeError: __init__() got an unexpected keyword argument 'tpus'
**Describe the bug**
Cannot fit N-Beats model to my data with a TPU in Google Colab.

**To Reproduce**
```
!pip install cloud-tpu-client==0.10 https://storage.googleapis.com/tpu-pytorch/wheels/torch_xla-1.9-cp37-cp37m-linux_x86_64.whl
!pip install torch==1.9.0+cu111 torchvision==0.10.0+cu111 torchtext==0.10.0 -f https://download.pytorch.org/whl/cu111/torch_stable.html
!pip install 'u8darts[torch]==0.17.1'
!pip install pyyaml==5.4.1
```
and then after restarting the runtime:
```
import numpy as np

from darts import TimeSeries
from darts.models import NBEATSModel

series = TimeSeries.from_values(np.random.random_sample((500,)))

trainset_size = 0.6
train, val = series.split_after(trainset_size)

model_nbeats = NBEATSModel(
  input_chunk_length=32,
  output_chunk_length=8,
  #generic_architecture=True,
  #num_stacks=32,
  #num_blocks=3,
  num_layers=4,
  layer_widths=256,
  n_epochs=150,
  nr_epochs_val_period=1,
  batch_size=128,
  pl_trainer_kwargs={
    "accelerator": "tpu",
    "tpus": [0]
  },
)
model_nbeats.fit(series=train, val_series=val, verbose=True) # Error here
```
I get the following error:
```
TypeError                                 Traceback (most recent call last)
[<ipython-input-4-1c57a1bae7f2>](https://localhost:8080/#) in <module>
     25   },
     26 )
---> 27 model_nbeats.fit(series=train, val_series=val, verbose=True)

6 frames
[/usr/local/lib/python3.7/dist-packages/darts/utils/torch.py](https://localhost:8080/#) in decorator(self, *args, **kwargs)
     68         with fork_rng():
     69             manual_seed(self._random_instance.randint(0, high=MAX_TORCH_SEED_VALUE))
---> 70             return decorated(self, *args, **kwargs)
     71 
     72     return decorator

[/usr/local/lib/python3.7/dist-packages/darts/models/forecasting/torch_forecasting_model.py](https://localhost:8080/#) in fit(self, series, past_covariates, future_covariates, val_series, val_past_covariates, val_future_covariates, trainer, verbose, epochs, max_samples_per_ts, num_loader_workers)
    769 
    770         return self.fit_from_dataset(
--> 771             train_dataset, val_dataset, trainer, verbose, epochs, num_loader_workers
    772         )
    773 

[/usr/local/lib/python3.7/dist-packages/darts/utils/torch.py](https://localhost:8080/#) in decorator(self, *args, **kwargs)
     68         with fork_rng():
     69             manual_seed(self._random_instance.randint(0, high=MAX_TORCH_SEED_VALUE))
---> 70             return decorated(self, *args, **kwargs)
     71 
     72     return decorator

[/usr/local/lib/python3.7/dist-packages/darts/models/forecasting/torch_forecasting_model.py](https://localhost:8080/#) in fit_from_dataset(self, train_dataset, val_dataset, trainer, verbose, epochs, num_loader_workers)
    913 
    914         # setup trainer
--> 915         self._setup_trainer(trainer, verbose, train_num_epochs)
    916 
    917         # TODO: multiple training without loading from checkpoint is not trivial (I believe PyTorch-Lightning is still

[/usr/local/lib/python3.7/dist-packages/darts/models/forecasting/torch_forecasting_model.py](https://localhost:8080/#) in _setup_trainer(self, trainer, verbose, epochs)
    522         self.trainer = (
    523             self._init_trainer(trainer_params=self.trainer_params, max_epochs=epochs)
--> 524             if trainer is None
    525             else trainer
    526         )

[/usr/local/lib/python3.7/dist-packages/darts/models/forecasting/torch_forecasting_model.py](https://localhost:8080/#) in _init_trainer(trainer_params, max_epochs)
    535             trainer_params_copy["max_epochs"] = max_epochs
    536 
--> 537         return pl.Trainer(**trainer_params_copy)
    538 
    539     @abstractmethod

[/usr/local/lib/python3.7/dist-packages/pytorch_lightning/trainer/connectors/env_vars_connector.py](https://localhost:8080/#) in insert_env_defaults(self, *args, **kwargs)
     36 
     37         # all args were already moved to kwargs
---> 38         return fn(self, **kwargs)
     39 
     40     return insert_env_defaults

TypeError: __init__() got an unexpected keyword argument 'tpus'
```

**Expected behavior**
I thought I could use TPU based on this [comment](https://github.com/unit8co/darts/issues/802#issuecomment-1044130508).

**System:**
 - Python version: 3.9.7
 - darts version: 0.17.1
 - pytorch-lightning  version: 1.5.10
- torch version: 1.9.0+cu111
- torch-xla version: 1.9
- torchaudio version: 0.10.0+cu111
- torchmetrics version: 0.7.2
- torchsummary version: 1.5.1
- torchtext version: 0.10.0
- torchvision version: 0.10.0+cu111

**Additional context**

- https://pytorch-lightning.readthedocs.io/en/stable/advanced/tpu.html#colab-tpus
- https://github.com/pytorch/xla/issues/3215

### PR DESCRIPTION
**Pull request by gsamaras**
Fixes #808 .

### Summary

As discussed, added a guide on how to use GPU and TPU with PyTorch Lightning. Also shows that a model uses CPU by default, based on an RNN example.

### Other Information

[off-topic] The documentation has a typo in Timeseries [here](https://unit8co.github.io/darts/generated_api/darts.timeseries.html?highlight=timeseries#darts.timeseries.TimeSeries.astype), which I corrected in this pull request.

### CODE REVIEW

File: docs/userguide/gpu_and_tpu_usage.md [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+0.004     Total estimated model params size (MB)
+
+Epoch 299: 100% 8/8 [00:00<00:00, 42.49it/s, loss=0.00285, v_num=logs]
+<darts.models.forecasting.rnn_model.RNNModel at 0x7fc75901cb10>
+```
+
+Now the model is ready to start predicting, which won't be shown here since it's included in the example linked in the start of this guide.
+
+## Use a GPU
+GPUs can dramatically improve the performance of your model in terms of processing time. We only need to instruct our model to use our machine's GPU through PyTorch Lighnting Trainer parameters, which are expressed as the `pl_trainer_kwargs` dictionary, like this:
```
Comment (hrzn) [inline docs/userguide/gpu_and_tpu_usage.md:83]: You could add a reference to the Pytorch Lightning doc (e.g. this page: https://pytorch-lightning.readthedocs.io/en/stable/common/trainer.html#accelerator) here.

**Review by hrzn — COMMENTED** [review]
This is great, thanks a lot :)
My only suggestion would be to add a link to PyTorch Lightning accelerator documentation, as it will provide reference documentation as to how to use accelerators.

File: docs/userguide/gpu_and_tpu_usage.md (hunk 1 above)
Comment (gsamaras) [reply docs/userguide/gpu_and_tpu_usage.md]: @hrzn yes, argh I knew I forgot sth, thanks! OK, so now I pushed the update in my cloned repo (gsamaras/darts). What I should do now? It seems like that change is already reflected in this pull request?

Comment (hrzn) [reply docs/userguide/gpu_and_tpu_usage.md]: Thanks! Basically there's nothing more to do on your side for now. We'll wait for the checks to finish and then merge your branch. Thanks for the contribution :)

**Review by hrzn — APPROVED** [review]

**Pull request merged**